friends/src/reducers/index.js

```diff
@@ -0,0 +1,72 @@
+
+import {
+    LOGIN_SUCCESS,
+    LOGIN_FAILURE,
+    LOGIN_START,
+    FETCH_FRIENDS_START,
+    FETCH_FRIENDS_SUCCESS,
+    FETCH_FRIENDS_FAILURE,
+    DELETE_FRIEND
+  } from "../actions";
+  
+
+const initialState = {
+    deletingFriend: false,
+    fetchingFriends: false,
+    friends: [],
+    loggingIn: false,
+    savingFriends: false,
+    updatingFriend: false,
+    error: null
+    // Array characters, Boolean fetching, null error.
+  };
+
+
+
+  export const rootReducer =(state=initialState, action) => {
+      switch(action.type) {
+          case LOGIN_START:
+            return {
+                ...state,
+                loggingIn: true
+            }
```

Comment:
Don't forget to set `error` to `null` as well for this action (to be consistent with your other actions too).